daonode/src/main/java/bisq/daonode/endpoints/SignedWitnessApi.java

```diff
@@ -0,0 +1,77 @@
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.daonode.endpoints;
+
+import bisq.core.account.witness.AccountAgeWitnessService;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+
+
+import bisq.daonode.DaoNodeRestApiApplication;
+import bisq.daonode.ServiceNode;
+import bisq.daonode.dto.ProofOfBurnDto;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.Application;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+
+/**
+ * Endpoint for getting the signed witness date from a given hash as hex string.
+ * Used for reputation system in Bisq 2.
+ * <a href="http://localhost:8082/api/v1/signed-witness/get-date/dd75a7175c7c83fe9a4729e36b85f5fbc44e29ae">Request with hash</a>
+ */
+@Slf4j
+@Path("/signed-witness")
+@Produces(MediaType.APPLICATION_JSON)
+@Tag(name = "Signed witness API")
+public class SignedWitnessApi {
+    private static final String DESC_HASH = "The hash of the signed account age witness as hex string";
+    private final ServiceNode serviceNode;
+
+    public SignedWitnessApi(@Context Application application) {
+        serviceNode = ((DaoNodeRestApiApplication) application).getServiceNode();
+    }
+
+    @Operation(description = "Request the signed witness date")
+    @ApiResponse(responseCode = "200", description = "The signed witness date",
+            content = {@Content(mediaType = MediaType.APPLICATION_JSON,
+                    schema = @Schema(allOf = ProofOfBurnDto.class))}
+    )
+    @GET
+    @Path("get-date/{hash}")
+    public Long getDate(@Parameter(description = DESC_HASH)
+                        @PathParam("hash")
+                        String hash) {
+        AccountAgeWitnessService accountAgeWitnessService = checkNotNull(serviceNode.getAccountAgeWitnessService());
+        return accountAgeWitnessService.getWitnessByHashAsHex(hash)
+                .map(accountAgeWitnessService::getWitnessSignDate)
+                .orElse(-1L);
```

Comment:
This enpoint's respnose header content-type = "application/json", but the response content-type can be "text/plain", i.e, -1 or a date.  A requester is seeing a "application/json" header in the response, but is getting a "text/plain" value instead.

Should these "text/plain" values be wrapped in json in the response?

Reply:
Maybe it does not matter, but my main question is "is it OK to send a raw long in the response when its content-type header is "application/json"?

Reply:
OK change to text_plain